### PR TITLE
docs(connector-sendgrid-mail): add readme.md (en-US)

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add SendGrid mail README.md (en-US).

## Testing

N/A

### CODE REVIEW

File: packages/connector-sendgrid-mail/README.md [hunk 1]
```diff
@@ -1,2 +1,86 @@
-### SendGrid Mail README
-placeholder
+# SendGrid mail connector
+
+SendGrid (also known as Twilio SendGrid) is a customer communication platform for transactional and marketing email. We can use its email sending function to send a "verification code" (or in other words, passcode).
+
+## Register and sign in SendGrid account
+
+Go to https://app.sendgrid.com/ to register a new account and sign in to SendGrid afterward. You may skip the registration step if you've already got an account.
```
Comment (wangsijie) [inline packages/connector-sendgrid-mail/README.md:7]: add a link?

**Review by wangsijie — APPROVED** [review]

File: packages/connector-sendgrid-mail/README.md [hunk 2]
```diff
@@ -1,2 +1,86 @@
-### SendGrid Mail README
-placeholder
+# SendGrid mail connector
+
+SendGrid (also known as Twilio SendGrid) is a customer communication platform for transactional and marketing email. We can use its email sending function to send a "verification code" (or in other words, passcode).
```
Comment (simeng-li) [inline packages/connector-sendgrid-mail/README.md:3]:
```suggestion
SendGrid (a.k.a. Twilio SendGrid) is a customer communication platform for transactional and marketing email. We can use its email sending function to send a "verification code".
```

Comment (simeng-li) [reply packages/connector-sendgrid-mail/README.md:3]: Should align the description sentence of your Twillos doc.  This is more straightforward.

File: packages/connector-sendgrid-mail/README.md (hunk 1 above)
Comment (simeng-li) [reply packages/connector-sendgrid-mail/README.md:7]: Simply: `Sign in or create a new account at XXXXX. `

File: packages/connector-sendgrid-mail/README.md [hunk 3]
```diff
@@ -1,2 +1,86 @@
...
+
+## Register and sign in SendGrid account
+
+Go to https://app.sendgrid.com/ to register a new account and sign in to SendGrid afterward. You may skip the registration step if you've already got an account.
+
+## Verify senders
+
+Senders are the email addresses where the emails containing verification codes come from. In order to send emails via the SendGrid mail server, you need to verify at least one sender.
```
Comment (simeng-li) [inline packages/connector-sendgrid-mail/README.md:11]:
```suggestion
Senders indicate the addresses our verification code email will be sent from.  In order to send emails via the SendGrid mail server, you need to verify at least one sender.
```

File: packages/connector-sendgrid-mail/README.md [hunk 4]
```diff
@@ -1,2 +1,86 @@
...
+
+Starting from the [SendGrid console page](https://app.sendgrid.com/), go to _Settings_ -> _Sender Authentication_ from the sidebar.
+
+Domain Authentication is recommended but it is not obligated. You can click _Get Started_ in _Authenticate Your Domain_ card and follow the guide.
```
Comment (simeng-li) [inline packages/connector-sendgrid-mail/README.md:15]:
```suggestion
Domain Authentication is recommended but not obligatory. You can click _Get Started_ in _Authenticate Your Domain_ card and follow the guide.
```

Comment (simeng-li) [reply packages/connector-sendgrid-mail/README.md:15]: image

What's the difference if I click on the Create a sender at the home page?

File: packages/connector-sendgrid-mail/README.md [hunk 5]
```diff
@@ -1,2 +1,86 @@
...
+
+By clicking the _Verify a Single Sender_ button in the panel, you are now focusing on a form requiring some critical information to create a sender. Follow the guide, fill out all these fields, and hit the _Create_ button.
+
+After the single sender is created, an email with a verification link should be sent to your sender's email address. Go to your mailbox, find the verification mail and finish verifying the single sender by clicking the link given in the email. You can now send email via SendGrid email connector using the sender you've just verified.
```
Comment (simeng-li) [inline packages/connector-sendgrid-mail/README.md:19]: Following the guide to link and verify a sender to SendGrid.

**Review by simeng-li — APPROVED** [review]
LGMT

File: packages/connector-sendgrid-mail/README.md [hunk 6]
```diff
@@ -1,2 +1,88 @@
...
+
+Fill out the `apiKey` field with the API Key created in **Create API keys** section.
+
+Fill out the `fromEmail` and `fromName` fields with the senders' _From Address_ and _Nickname_. You can find the sender's details on the [_Sender Management_ page](https://mc.sendgrid.com/senders).
+
+You can add multiple SendGrid mail connector templates for different cases. Here is an example of adding a single template:
+    - Fill out the `subject` field, which works as the title of emails.
+    - Fill out the `content` field with arbitrary string-typed contents. Do not forget to leave the `{{code}}` placeholder for the random passcode.
+    - Fill out `usageType` field with either `Register`, `SignIn` or `Test` for different use cases.
+    - Fill out `type` field with either `text/plain` or `text/html` for different types of content.
```
Comment (IceHe) [inline packages/connector-sendgrid-mail/README.md:41]: Strange indents?

```suggestion
You can add multiple SendGrid mail connector templates for different cases. Here is an example of adding a single template:

- Fill out the `subject` field, which works as the title of emails.
- Fill out the `content` field with arbitrary string-typed contents. Do not forget to leave the `{{code}}` placeholder for the random passcode.
- Fill out `usageType` field with either `Register`, `SignIn` or `Test` for different use cases.
- Fill out `type` field with either `text/plain` or `text/html` for different types of content.
```

Comment (darcyYe) [reply packages/connector-sendgrid-mail/README.md:41]: Lol, you commented right after I merged this PR.
Don't worry, I will double-check all connector-related layouts before launch.

**Pull request merged**